src/main/scala/com/github/pathikrit/metarest/Resource.scala

```diff
@@ -1,74 +1,55 @@
 package com.github.pathikrit.metarest
 
-import scala.annotation.StaticAnnotation
-import scala.reflect.macros.blackbox
-
-class Resource extends StaticAnnotation {
-  def macroTransform(annottees: Any*): Any = macro Resource.impl
-}
+import scala.collection.immutable.Seq
+import scala.collection.mutable
+import scala.annotation.{StaticAnnotation, compileTimeOnly}
+import scala.meta._
 
 class get extends StaticAnnotation
 class put extends StaticAnnotation
 class post extends StaticAnnotation
 class patch extends StaticAnnotation
 
-object Resource {
-  def impl(c: blackbox.Context)(annottees: c.Expr[Any]*): c.Expr[Any] = {
-    import c.universe._
-
-    def withCompileError[A](msg: String)(block: => A): A = try block catch {
-      case e: MatchError => c.abort(c.enclosingPosition, s"MetaRest: $msg; Got: $e")
-    }
-
-    def getClassAndFields(classDecl: ClassDef) = withCompileError("must annotate a case class") {
-      val q"case class $className(..$fields) extends ..$bases { ..$body }" = classDecl
-      (className, fields)
-    }
-
-    def generateModels(fields: List[ValDef]) = {
-      val fieldsWithAnnotations = fields.map(_.mods.annotations).zip(fields)
-      val newFields = fieldsWithAnnotations flatMap { case (annotations, field) =>
-        annotations.map(_ -> field) collect {
-          case (q"new patch", q"$accessor val $vname: $tpe") => "patch" -> q"$accessor val $vname: Option[$tpe] = None"
-          case (q"new $annotation", f) if Set("get", "post", "put") contains annotation.toString => annotation.toString -> f.duplicate
-        }
-      }
-      newFields.groupBy(_._1) map { case (annotation, values) =>
-        val (className, classFields) = (TypeName(annotation.capitalize), values.map(_._2))
-        q"case class $className(..$classFields)"
-      }
+@compileTimeOnly("@metarest.Resource not expanded")
```

Comment:
I'm not sure this annotation is necessary. I think it's added automatically by paradise.